test/nfs-testdriver.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/testpatterns"
+	"k8s.io/kubernetes/test/e2e/storage/testsuites"
+)
+
+type nfsDriver struct {
+	driverInfo testsuites.DriverInfo
+	manifests  []string
+}
+
+var NFSdriver = InitNFSDriver
+
+type nfsVolume struct {
+	serverIP  string
+	serverPod *v1.Pod
+	f         *framework.Framework
+}
+
+// initNFSDriver returns nfsDriver that implements TestDriver interface
+func initNFSDriver(name string, manifests ...string) testsuites.TestDriver {
+	return &nfsDriver{
+		driverInfo: testsuites.DriverInfo{
+			Name:        name,
+			MaxFileSize: testpatterns.FileSizeLarge,
+			SupportedFsType: sets.NewString(
+				"", // Default fsType
+			),
+			Capabilities: map[testsuites.Capability]bool{
+				testsuites.CapPersistence: true,
+				testsuites.CapExec:        true,
+			},
+		},
+		manifests: manifests,
+	}
+}
+
+func InitNFSDriver() testsuites.TestDriver {
+
+	return initNFSDriver("csi-nfsplugin",
+		"csi-attacher-nfsplugin.yaml",
+		"csi-attacher-rbac.yaml",
+		"csi-nodeplugin-nfsplugin.yaml",
+		"csi-nodeplugin-rbac.yaml")
+
+}
+
+var _ testsuites.TestDriver = &nfsDriver{}
+var _ testsuites.PreprovisionedVolumeTestDriver = &nfsDriver{}
+var _ testsuites.PreprovisionedPVTestDriver = &nfsDriver{}
+
+func (n *nfsDriver) GetDriverInfo() *testsuites.DriverInfo {
+	return &n.driverInfo
+}
+
+func (n *nfsDriver) SkipUnsupportedTest(pattern testpatterns.TestPattern) {
+	if pattern.VolType == testpatterns.DynamicPV {
+		framework.Skipf("NFS Driver does not support dynamic provisioning -- skipping")
```

Comment:
in the future it may be cool to add something like the nfs-client provisioner to the sample driver, and then we can unlock these test cases as well

Reply:
+1